src/error.rs

```diff
@@ -85,6 +85,14 @@ pub enum Error {
 
     #[error("Cannot convert quantity [{value}] to Java heap value with unit [{target_unit}].")]
     CannotConvertToJavaHeapValue { value: String, target_unit: String },
+    #[error("Conversion error: [message]")]
+    ConversionError { message: String },
+
+    #[error("Given container name: '{container_name}'. Reason: {violation}")]
+    InvalidContainerName {
+        container_name: String,
+        violation: String,
```

Comment:
Happy to do this. The function's signature would have to be changed to something like `pub fn is_rfc_1123_label(value: &str) -> Result<(), Vec<Error>> {`.

Where `Error` is
```rust
#[derive(Debug, thiserror::Error)]
pub enum Error {
    #[error("Maximum length of {max_len} characters exceeded")]
    LengthExceeded { max_len: usize },
    #[error("{formatting_hint}")]
    InvalidFormat { formatting_hint: String },
}
```

Unfortunately thiserror does not support multiple error causes.

```
    #[error("container name {source}")]
    InvalidContainerName {
        #[from]
        source: Vec<crate::validation::Error>,
    },
```

Not sure if worth doing in this MR. I'd stick to current convention.

Reply:
No, it doesn't at the moment. https://github.com/dtolnay/thiserror/issues/126
Alternatively, a single error with all violations written in it will work. The cost is obvious, inability to distinguish which validation failed in code.

Reply:
Yeah I'd be ok with punting on this for now.